lib/_http_agent.js

```diff
@@ -127,7 +129,15 @@ function Agent(options) {
     const requests = this.requests[name];
     if (requests && requests.length) {
       const req = requests.shift();
-      setRequestSocket(this, req, socket);
+      const reqAsyncRes = req[kRequestAsyncResource];
+      if (reqAsyncRes) {
```

Comment:
Probably worth a comment to explain that this codepath is for

Reply:
Makes sense. Added a comment for that.